api/api/clubs/permissions.py

```diff
@@ -0,0 +1,12 @@
+from django.db.models import Q
+from rest_framework import permissions
+
+
+class IsPositionHolder(permissions.BasePermission):
+    """Custom permission to allow only position holders of respective club"""
+
+    def has_object_permission(self, request, view, obj):
+        allowed_roles = obj.roles.filter(Q(name='Coordinator') | Q(name='Co-Coordinator'))
+
+        # return true when allowed_roles and users roles have common elements
+        return (allowed_roles & request.user.roles.all()).exists()
```

Comment:
A small optimization: instead of considering all the roles of the user, we could consider just the roles of the user for that club.
```python
request.user.roles.filter(club__id=obj.id)
```

Just try it out.